validator_client/src/duties_service.rs

```diff
@@ -102,15 +102,26 @@ impl DutiesStore {
             .count()
     }
 
+    fn attesters_for_epoch(&self, epoch: Epoch) -> Vec<PublicKey> {
+        self.store
+            .read()
+            .iter()
+            .flat_map(|(_validator_pubkey, validator_map)| {
+                validator_map
+                    .get(&epoch)
+                    .map(|duties| duties.validator_pubkey.clone())
+            })
+            .collect()
+    }
+
     fn block_producers(&self, slot: Slot, slots_per_epoch: u64) -> Vec<PublicKey> {
+        let epoch = slot.epoch(slots_per_epoch);
```

Comment:
I've moved this one out of the filter_map function for code optimisation (though I presume the compiler already moved it out when compiling...)